.github/workflows/update-reproducibles.yml

```diff
@@ -0,0 +1,42 @@
+name: Update reproducible requirements
+on:
+  pull_request:
+    branches: [develop]
+    paths:
+      - requirements/base.txt # run this GHA only if requirements file is changed
+
+jobs:
+  update-reqs:
+    permissions:
+      pull-requests: write
+      contents: write
+    if: github.repository_owner == 'aws'
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            python: 3.11
+            target: update-reproducible-linux-reqs
+          - os: macos-latest
+            python: 3.8
+            target: update-reproducible-mac-reqs
+          - os: windows-latest
+            python: 3.8
+            target: update-reproducible-win-reqs
+      max-parallel: 1
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }}
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python }}
+      - run: make ${{ matrix.target }}
+      - name: Push changes
+        run: |
+          git config --global user.email "action@github.com"
+          git config --global user.name "GitHub Action"
+          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/${{ github.repository }}
+          git commit -am "Update reproducibles: ${{ matrix.target }}" || echo "nothing to commit"
```

Comment:
Just a general comment, butI think I had the same conversation with a team member about auto generating a JSON schema file, and whether or not this should be a message instead to manually run the generation rules, or something that Github Actions will do for us in the same PR.

The wrench that is thrown here is that Windows repro generation would in theory need to happen on a Windows machine.

Reply:
The reason we put these updates into GHA is purely needing a Windows machine to update reproducibles. If these were not OS specific (like we can generate one and use it for every installer), then I would keep the existing flow where we will have a Makefile target which can be run on any OS so that contributors can run it after they update any dependency that we have.